### PR TITLE
Fix CI config for concurrency in master branch

### PR DESCRIPTION
**Pull request by Ao-senXiong**
I checked the cancelled job, the name of reference for master branch is `CI tests-refs/heads/master' exists`. So instead of using `master/`, we should use `master`

### CODE REVIEW

**Comment by Ao-senXiong** [comment]
This config cancel previous commit.

**Comment by wmdietl** [comment]
> I checked the cancelled job, the name of reference for master branch is `CI tests-refs/heads/master' exists`. So instead of using `master/`, we should use `master`

Should it be `heads/master` to avoid conflicts with a branch that contains `master`?

**Comment by Ao-senXiong** [comment]
> heads/master

Sure, updated.

**Comment by wmdietl** [comment]
Right now it looks like tests for [0b1fb14](https://github.com/eisop/checker-framework/pull/872/commits/0b1fb140f3c0442134455f47a1b7518bc79c618b) and [d3ce3d1](https://github.com/eisop/checker-framework/pull/872/commits/d3ce3d126741110fae68a3cd4d829f9a29be4eee) are both running.
Maybe the cancelling didn't happen yet, because no GitHub Actions for the newest commit was executed?

**Comment by Ao-senXiong** [comment]
> Right now it looks like tests for [0b1fb14](https://github.com/eisop/checker-framework/pull/872/commits/0b1fb140f3c0442134455f47a1b7518bc79c618b) and [d3ce3d1](https://github.com/eisop/checker-framework/pull/872/commits/d3ce3d126741110fae68a3cd4d829f9a29be4eee) are both running. Maybe the cancelling didn't happen yet, because no GitHub Actions for the newest commit was executed?

This is strange. It should cancel previous CI when there is new job scheduled. Let me do an empty commit and see what happen.

**Comment by Ao-senXiong** [comment]
Looks like it cancelled [d3ce3d1](https://github.com/eisop/checker-framework/pull/872/commits/d3ce3d126741110fae68a3cd4d829f9a29be4eee) but not [0b1fb14](https://github.com/eisop/checker-framework/pull/872/commits/0b1fb140f3c0442134455f47a1b7518bc79c618b)

**Comment by Ao-senXiong** [comment]
I got email for says `.github/workflows/ci.yml: No jobs were run
`
[Error](https://github.com/eisop/checker-framework/actions/runs/10479749317/workflow)
Invalid type for `on`

**Comment by Ao-senXiong** [comment]
let's wait util other PRs finished and try again. Looks like there is no enough workers.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,9 +8,9 @@ on:
 
 # Configure GitHub Actions cancel in progress workflow to avoid redundant runs in pull requests.
 # See: https://docs.github.com/en/enterprise-cloud@latest/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs
-concurrency:
+  concurrency:
```
Comment (wmdietl) [inline .github/workflows/ci.yml:11]: Maybe this is the problem with `on`? The indentation makes this belong to the `on` from line 3?

Comment (Ao-senXiong) [reply .github/workflows/ci.yml:11]: It is. Thanks!

**Comment by Ao-senXiong** [comment]
Now I think it works correctly.

**Review by wmdietl — APPROVED** [review]
Thanks! Let's see whether it works as expected.

**Pull request merged**